src/core/thread/mle.cpp

```diff
@@ -730,7 +735,7 @@ void Mle::HandleNetifStateChanged(uint32_t aFlags, void *aContext)
 
 void Mle::HandleNetifStateChanged(uint32_t aFlags)
 {
-    VerifyOrExit((aFlags & (OT_IP6_ADDRESS_ADDED | OT_IP6_ADDRESS_REMOVED)) != 0, ;);
```

Comment:
The latter portion of this method (the `switch` statement below) needs to handle both `OT_IP6_ADDRESS_ADDED` and `OT_IP6_ADDRESS_REMOVED` events.